craft_application/application.py

```diff
@@ -83,7 +86,12 @@ class AppMetadata:
     ConfigModel: type[_config.ConfigModel] = _config.ConfigModel
 
     ProjectClass: type[models.Project] = models.Project
-    BuildPlannerClass: type[models.BuildPlanner] = models.BuildPlanner
+    BuildPlannerClass: Annotated[
+        type[models.BuildPlanner] | None,
+        pydantic.Field(
```

Comment:
what does annotating this as a pydantic.Field do?

Reply:
In auto-generated documentation for the AppMetadata class it'll use those pydantic docs - in this case the deprecation warning.